### PR TITLE
Feature/25 phone validation

### PR DESCRIPTION
**Pull request by IrvingJVR-MX**
Added verification by phone number

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -104,6 +104,24 @@ dependencies {
...
+    implementation 'com.hbb20:ccp:2.6.1'
+    implementation 'androidx.browser:browser:1.4.0'
+
+
+    // Dependency to include Maps SDK for Android
+    implementation 'com.google.android.gms:play-services-maps:18.1.0'
+    implementation 'com.google.android.gms:play-services-location:20.0.0'
+
+    //country code picker
+    implementation 'com.hbb20:ccp:2.6.1'
```
Comment (jmacboy) [inline app/build.gradle:109]: you have the libraries declared twice

File: app/src/main/java/com/jalasoft/routesapp/data/remote/managers/AuthFirebaseManager.kt [hunk 2]
```diff
@@ -25,6 +29,44 @@ class AuthFirebaseManager(private val auth: FirebaseAuth) : AuthFirebaseDataSour
         }
     }
 
+    override suspend fun sendPhoneNumberCode(phoneNumber: String, activity: Activity, mCallBack: PhoneAuthProvider.OnVerificationStateChangedCallbacks): Response<String> {
+        return try {
+            Firebase.auth.setLanguageCode(Locale.getDefault().language)
+            val options = PhoneAuthOptions.newBuilder(Firebase.auth)
+                .setPhoneNumber(phoneNumber)
+                .setActivity(activity)
+                .setTimeout(60L, TimeUnit.SECONDS)
```
Comment (jmacboy) [inline app/src/main/java/com/jalasoft/routesapp/data/remote/managers/AuthFirebaseManager.kt]: set this timeout on a constant variable

File: app/src/main/java/com/jalasoft/routesapp/ui/auth/phoneAuthentication/view/PhoneAuthenticationFragment.kt [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+        textInputLayout.addView(codeEditText)
+
+        val constraintSet = ConstraintSet()
+        constraintSet.clone(constraintLayout)
+
+        constraintLayout.addView(textInputLayout)
+        return constraintLayout
+    }
+
+    private fun Int.toDp(context: Context): Int = TypedValue.applyDimension(
```
Comment (jmacboy) [inline app/src/main/java/com/jalasoft/routesapp/ui/auth/phoneAuthentication/view/PhoneAuthenticationFragment.kt]: if this is an Utility to convert from pixels to DP, it should be moved to its own utilities file

**Review by jmacboy — CHANGES_REQUESTED** [review]

File: app/build.gradle (hunk 1 above)
Comment (IrvingJVR-MX) [reply app/build.gradle:109]: done

File: app/src/main/java/com/jalasoft/routesapp/data/remote/managers/AuthFirebaseManager.kt (hunk 2 above)
Comment (IrvingJVR-MX) [reply app/src/main/java/com/jalasoft/routesapp/data/remote/managers/AuthFirebaseManager.kt]: done

File: app/src/main/java/com/jalasoft/routesapp/ui/auth/phoneAuthentication/view/PhoneAuthenticationFragment.kt (hunk 3 above)
Comment (IrvingJVR-MX) [reply app/src/main/java/com/jalasoft/routesapp/ui/auth/phoneAuthentication/view/PhoneAuthenticationFragment.kt]: done

File: app/src/main/java/com/jalasoft/routesapp/util/ConverterFromPixelToDpUtil.kt [hunk 4]
```diff
@@ -0,0 +1,12 @@
+package com.jalasoft.routesapp.util
+
+import android.content.Context
+import android.util.TypedValue
+
+object ConverterFromPixelToDpUtil {
```
Comment (jmacboy) [inline app/src/main/java/com/jalasoft/routesapp/util/ConverterFromPixelToDpUtil.kt:6]: Maybe DisplayUtils or ScreenUtils could be more general and reusable. Still nice job

**Review by jmacboy — APPROVED** [review]

**Pull request merged**